src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -1145,14 +1145,14 @@ internal unsafe void Write(byte[] array, int offset, int count, int timeout)
             if (_inBreak)
                 throw new InvalidOperationException(SR.In_Break_State);
             if (array == null)
-                throw new ArgumentNullException(nameof(array), SR.ArgumentNull_Array);
+                throw new ArgumentNullException(nameof(array));
             if (offset < 0)
                 throw new ArgumentOutOfRangeException(nameof(offset), SR.ArgumentOutOfRange_NeedPosNum);
             if (count < 0)
                 throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_NeedPosNum);
             if (count == 0) return; // no need to expend overhead in creating asyncResult, etc.
             if (array.Length - offset < count)
-                throw new ArgumentException(nameof(count), SR.ArgumentOutOfRange_OffsetOut);
+                throw new ArgumentException(SR.Argument_InvalidOffLen);
```

Comment:
Every other place where `if (array.Length - offset < count)` is being checked is using `SR.Argument_InvalidOffLen` for the message, so use the same message here for consistency.

This was the only place where `SR.ArgumentOutOfRange_OffsetOut` was being used and the message doesn't make sense here (`"Either offset did not refer to a position in the string, or there is an insufficient length of destination character array."`).